dev/tools/controllerbuilder/template/externalresource.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package template
+
+const ExternalResourceTemplate = `
+package {{.Service}}
+
+import (
+	"fmt"
+	"strings"
+)
+
+const (
+	// TODO(user): Add service domain
+	serviceDomain = "//{{.KindToLower}}.googleapis.com"
+)
+
+// TODO(user): Define resource identity
+type {{.Kind}}Identity struct {
+	project    string
+	location   string
+	{{.KindToLower}} string
```

Comment:
I think we want to change the KindToLower to `kind` and `proto-source`
more detail in b/359404149. Can be in follow-up PRs

Reply:
I can take that task if needed and follow up in another PR